### PR TITLE
Move registry interop to shared CoreLib partition

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegCreateKeyEx.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegCreateKeyEx.cs]: All new files are copies from corefx

File: src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegEnumKeyEx.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    internal partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, CharSet = CharSet.Unicode, BestFitMapping = false, EntryPoint = "RegEnumKeyExW")]
+        internal static extern unsafe int RegEnumKeyEx(
+            SafeRegistryHandle hKey,
+            int dwIndex,
+            char[] lpName,
+            ref int lpcbName,
+            int[] lpReserved,
+            [Out]StringBuilder lpClass,
```
Comment (Anipik) [inline src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegEnumKeyEx.cs]: nit: should we use ```out``` keyword here ?

File: src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegQueryInfoKey.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using System.Text;
+
+internal partial class Interop
+{
+    internal partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, CharSet = CharSet.Unicode, BestFitMapping = false, EntryPoint = "RegQueryInfoKeyW")]
+        internal static extern int RegQueryInfoKey(
+            SafeRegistryHandle hKey,
+            [Out]StringBuilder lpClass,
```
Comment (Anipik) [inline src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegQueryInfoKey.cs]: should we use ```out``` keyword here

File: src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegEnumKeyEx.cs (hunk 2 above)
Comment (jkotas) [reply src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegEnumKeyEx.cs]: `[Out]` is correct here. `out` controls how the argument passed at language/runtime level. `[Out]` controls how the argument is marshaled.

**Pull request merged**